ledger/src/blockstore.rs

```diff
@@ -1476,7 +1511,29 @@ impl Blockstore {
             shred.reference_tick(),
             &data_index,
         );
+        if is_recovered || is_repaired {
+            let mut slot_stats = self.slot_stats.lock().unwrap();
+            let mut e = slot_stats.1.entry(slot_meta.slot).or_default();
+            if is_repaired {
+                e.num_repaired += 1;
+            } else if is_recovered {
+                e.num_recovered += 1;
+            }
+        }
         if slot_meta.is_full() {
+            let (num_repaired, num_recovered) = {
+                let mut slot_stats = self.slot_stats.lock().unwrap();
+                if let Some(e) = slot_stats.1.remove(&slot_meta.slot) {
+                    if solana_sdk::timing::timestamp().saturating_sub(slot_stats.0) > 30 * 1000 {
+                        let root = self.last_root();
+                        slot_stats.1.retain(|slot, _stats| *slot > root);
```

Comment:
I do not know how big we expect this hash-map to be. Might be better to use `BTreeMap::split_off` instead.
Maybe, instead of cleaning up every 30 seconds, also condition this purging logic on the size of the `HashMap`.